aws_lambda_powertools/event_handler/openapi/types.py

```diff
@@ -2,10 +2,12 @@
 
 import types
 from enum import Enum
-from typing import Any, Callable, Dict, Set, Type, TypedDict, Union
+from typing import TYPE_CHECKING, Any, Callable, Dict, Set, Type, TypedDict, Union
 
 from pydantic import BaseModel
-from typing_extensions import NotRequired
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired
```

Comment:
Not sure why ruff raised new TCH002 errors for files we've checked before in the previous PRs. 😕